packages/relay/src/lib/clients/mirrorNodeClient.ts

```diff
@@ -1273,4 +1285,68 @@ export class MirrorNodeClient {
     }
     return result;
   }
+
+  /**
+   * Retrieves and processes transaction record metrics from the mirror node based on the provided transaction ID.
+   *
+   * @param {string} transactionId - The ID of the transaction for which the record is being retrieved.
+   * @param {string} callerName - The name of the caller requesting the transaction record.
+   * @param {string} requestId - The unique identifier for the request, used for logging and tracking.
+   * @param {string} txConstructorName - The name of the transaction constructor associated with the transaction.
+   * @param {string} operatorAccountId - The account ID of the operator, used to calculate transaction fees.
+   * @returns {Promise<{transactionFee: number;} | undefined>} - An object containing the transaction fee if available, or `undefined` if the transaction record is not found.
+   */
+  public async getTransactionRecordMetrics(
+    transactionId: string,
+    callerName: string,
+    requestId: string,
+    txConstructorName: string,
+    operatorAccountId: string,
+  ): Promise<ITransactionRecordMetric | undefined> {
+    const formattedRequestId = formatRequestIdMessage(requestId);
+
+    this.logger.trace(
+      `${formattedRequestId} Get transaction record via mirror node: transactionId=${transactionId}, txConstructorName=${txConstructorName}, callerName=${callerName}`,
+    );
+
+    // poll mirror node to get transaction record
+    const transactionRecords = await this.repeatedRequest(
+      this.getTransactionById.name,
+      [transactionId, 0],
+      this.MIRROR_NODE_GET_CONTRACT_RETRIES,
+      formattedRequestId,
+    );
+
+    if (!transactionRecords) {
+      this.logger.warn(
+        `${formattedRequestId} No transaction record retrieved: transactionId=${transactionId}, txConstructorName=${txConstructorName}, callerName=${callerName}`,
+      );
+    } else {
+      const transactionRecord: IMirrorNodeTransactionRecord = transactionRecords.transactions.find(
+        (tx: any) => tx.transaction_id === formatTransactionId(transactionId),
+      );
+
```

Comment:
What happens when `transactionRecords.transactions` doesn't contain the `transactionId` you are looking for? This should be handled:
```typescript
      const transactionRecord: IMirrorNodeTransactionRecord = transactionRecords.transactions.find(
        (tx: any) => tx.transaction_id === formatTransactionId(transactionId),
      );
      if (!transactionRecord) {
        // handle here
      }
```

Reply:
The `transactionRecords` array is a response from `MirrorNodeClient.getTransactionById()`, where the ID is the `transactionId`. Therefore, the `transactionRecords` array should contain at least one transaction with the specified `transactionId`. If `transactionRecords` does not include a transaction with the given `transactionId`, it indicates that `transactionRecords` was returned as `null` from `repeatedRequest`, which has already been handled earlier.